### PR TITLE
improve fp16 tftrt prediction

### PR DESCRIPTION
**Pull request by jjsjann123**
delay fp32 to fp16 conversion to reduce accumulated rounding error

### CODE REVIEW

**Comment by jjsjann123** [comment]
Hope that we have not freeze the code for 1.7 yet. Otherwise I'll do a PR to master @aaroey

**Comment by aaroey** [comment]
I'll wait for @gunan's opinion.

**Comment by gunan** [comment]
If this can be reviewed in the next 2 hours I can still get it into 1.7

File: tensorflow/contrib/tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -547,6 +547,19 @@ class Converter {
...
+TRT_ShapedWeights ConvertFP32ToFP16(Converter& ctx,
+                                    const TRT_ShapedWeights& weights_src) {
+  auto dtype_new = tensorflow::DataType::DT_HALF;
+  TRT_ShapedWeights weights =
+      ctx.get_temp_weights(dtype_new, weights_src.shape_);
+  const float* src = static_cast<const float*>(weights_src.GetValues());
+  Eigen::half* dst = const_cast<Eigen::half*>(
+      static_cast<Eigen::half const*>(weights.GetValues()));
+  for (int64_t i = 0; i < weights_src.count(); i++) {
+    dst[i] = Eigen::half_impl::float_to_half_rtne(src[i]);
```
Comment (aaroey) [inline tensorflow/contrib/tensorrt/convert/convert_nodes.cc]: How large is the data to transform? Does this hurt performance?

Comment (jjsjann123) [reply tensorflow/contrib/tensorrt/convert/convert_nodes.cc]: This is only done at conversion time. Would not have any impact on inference.

**Review by aaroey — APPROVED** [review]

**Comment by aaroey** [comment]
Ping @gunan to see if now is still a good time.

**Review by gunan — APPROVED** [review]

**Comment by jjsjann123** [comment]
@aaroey Sure. Will do as follow up PRs to master.

**Comment by aaroey** [comment]
@jjsjann123 That would be great, thanks!

**Comment by suke27** [comment]
it that tools for convert float 32 model to float 16 model?

**Pull request merged**